tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -841,10 +857,105 @@ def test_bkt_creation_ios_after_rgw_restart(self):
+        total_parts = HA_CFG["5gb_mpu_data"]["total_parts"]
+        part_numbers = list(range(1, total_parts + 1))
+        self.system_random.shuffle(part_numbers)
+        multipart_obj_path = os.path.join(self.test_dir_path, "test_42257_file")
+        download_path = os.path.join(self.test_dir_path, "test_42257_file_download")
+        part_numbers = list(range(1, total_parts + 1))
+        self.system_random.shuffle(part_numbers)
+        output = Queue()
+        parts_etag = list()
+        event = threading.Event()  # Event to be used to send intimation of cluster restart
+
+        self.log.info("Step 1: Start multipart upload of 5GB object in background")
+        args = {'s3_data': self.iam_user, 'bucket_name': self.bucket_name,
+                'object_name': self.object_name, 'file_size': file_size, 'total_parts': total_parts,
+                'multipart_obj_path': multipart_obj_path, 'part_numbers': part_numbers,
+                'parts_etag': parts_etag, 'output': output}
+        thread = threading.Thread(target=self.ha_obj.start_random_mpu, args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        self.log.info("Step 1: Started multipart upload of 5GB object in background")
+
+        time.sleep(self.delay)
+        self.log.info("Step 2: Perform Single rgw Process Restart")
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj,
+                                            pod_prefix=const.SERVER_POD_NAME_PREFIX,
+                                            container_prefix=const.RGW_CONTAINER_NAME,
+                                            process=self.rgw_process, check_proc_state=True)
+        assert_utils.assert_true(resp, "Failure observed during process restart/recovery")
+        self.log.info("Step 2: rgw restarted and recovered successfully")
+
+        self.log.info("Step 3: Wait for all parts to upload.")
+        thread.join()
+        responses = tuple()
+        while len(responses) < 4:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+
+        if not responses:
+            assert_utils.assert_true(False, "Background process failed to do multipart upload")
+
+        exp_failed_parts = responses[0]
+        failed_parts = responses[1]
+        parts_etag = responses[2]
+        mpu_id = responses[3]
+        self.log.debug("Responses received from background process:\nexp_failed_parts: "
+                       "%s\nfailed_parts: %s\nparts_etag: %s\nmpu_id: %s", exp_failed_parts,
+                       failed_parts, parts_etag, mpu_id)
+        if len(exp_failed_parts) == 0 and len(failed_parts) == 0:
+            self.log.info("All the parts are uploaded successfully")
+        else:
+            assert_utils.assert_true(False, "Failed to upload parts when during rgw restart."
```

Comment:
aren't we expecting failure during rgw restart?

Reply:
By default, retry attempts are 6. And the upload should be successful when retried.

Reply:
Then we need to set event as well to get right responses